src/block/compress.rs

```diff
@@ -177,7 +177,7 @@ fn count_same_bytes(input: &[u8], cur: &mut usize, source: &[u8], candidate: usi
     // compare 4 bytes block
     #[cfg(target_pointer_width = "64")]
     {
-        if *cur + 4 <= input_end {
```

Comment:
cur and input_end are usize, they are unaffected by the issue

Reply:
Right, but since we're dealing with under/overflow style problems I thought it would be best to fix all of them. Theoretically an allocation at the end of the address space could lead to *cur + 4 to overflow.